### PR TITLE
fix: Change ruff errors to warnings and fix config saving

### PR DESCRIPTION
**Pull request by mattrunyon**
Changed all except syntax errors to warning underlines instead of error underline since they are not actual errors.

Also changed redux to not save if the config matches the default. This should fix an edge case where a user opens the settings editor just to see what the config is, then hits "Save". Previously, this would save the current defaults as their settings and then if the defaults changed they would not get updated settings. This way, if a user was on defaults and the default changes they will get the new default.

Also added a little bit of debug logging for Ruff.

### CODE REVIEW

File: packages/code-studio/src/settings/EditorSectionContent.tsx [hunk 1]
```diff
@@ -51,15 +51,24 @@ export function EditorSectionContent(): JSX.Element {
     (newValue: Record<string, unknown>) => {
       dispatch(
         updateNotebookSettings({
-          python: { linter: { ...ruffSettings, config: newValue } },
+          python: {
+            linter: {
+              ...ruffSettings,
+              config:
+                JSON.stringify(newValue) ===
```
Comment (bmingles) [inline packages/code-studio/src/settings/EditorSectionContent.tsx:58]: Is the argument mismatch in the JSON.stringify calls intentional?

File: packages/code-studio/src/settings/EditorSectionContent.tsx [hunk 2]
```diff
@@ -51,15 +51,24 @@ export function EditorSectionContent(): JSX.Element {
...
+          },
         })
       );
       MonacoProviders.setRuffSettings(newValue);
     },
     [dispatch, ruffSettings]
   );
 
-  const [val, setVal] = useState(JSON.stringify(ruffConfig, null, 2));
+  const val = JSON.stringify(ruffConfig, null, 2);
```
Comment (bmingles) [inline packages/code-studio/src/settings/EditorSectionContent.tsx:71]: Should we memoize this?

**Review by bmingles — CHANGES_REQUESTED** [review]
Looks good overall. Left 2 suggestion / questions

File: packages/code-studio/src/settings/EditorSectionContent.tsx (hunk 1 above)
Comment (mattrunyon) [reply packages/code-studio/src/settings/EditorSectionContent.tsx:58]: Yes. I want to check if the value was reset to default here, but the objects won't be the same. The other option was to do this check in the settings modal before onSave is called and then call with null or undefined to indicate the default.

File: packages/code-studio/src/settings/EditorSectionContent.tsx (hunk 2 above)
Comment (mattrunyon) [reply packages/code-studio/src/settings/EditorSectionContent.tsx:71]: I don't think it matters. The object isn't that big and there's still some overhead to do the string comparison for memoization. Also the component shouldn't re-render often.

File: packages/code-studio/src/settings/EditorSectionContent.tsx (hunk 1 above)
Comment (bmingles) [reply packages/code-studio/src/settings/EditorSectionContent.tsx:58]: Oh, I misread this. I thought your were passing in `undefined, 2` to one of the JSON.stringify calls

**Review by bmingles — APPROVED** [review]
LGTM

**Pull request merged**